tiledb/sm/filesystem/azure.cc

```diff
@@ -689,7 +610,7 @@ Status Azure::blob_size(const URI& uri, uint64_t* const nbytes) const {
     if (response.Blobs.empty()) {
       error_message = "Blob does not exist.";
     } else {
-      *nbytes = static_cast<uint64_t>(response.Blobs[0].BlobSize);
+      nbytes = static_cast<uint64_t>(response.Blobs[0].BlobSize);
```

Comment:
```suggestion
      return static_cast<uint64_t>(response.Blobs[0].BlobSize);
```

And remove the `nbytes` variable.

---

Off-topic, but I think that this method should return an `int64_t` instead of `uint64_t`, and that in general unsigned integers should not be used to represent quantities. The C API does this a lot and it has caused difficulties when interacting with environments that primarily use signed integers (like .NET and Go), and signed integers are easier to work with in some cases. Changing this now at the public API layer would be a breaking change, but it's something for us to consider in future additions.